tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -3017,3 +3018,170 @@ def call(self, inputs, state):
 
       new_state = rnn_cell_impl.LSTMStateTuple(new_c, new_h)
       return new_h, new_state
+
+
+class NLSTMCell(rnn_cell_impl.RNNCell):
+  """Nested LSTM Cell. Adapted from `rnn_cell_impl.LSTMCell`
+
+  The implementation is based on:
+    https://arxiv.org/abs/1801.10308
+    JRA. Moniz, D. Krueger.
+    "Nested LSTMs"
+    ACML, PMLR 77:530-544, 2017
+  """
+
+  def __init__(self, num_units, depth, forget_bias=1.0,
+               state_is_tuple=True, activation=None, reuse=None, name=None):
+    """Initialize the basic NLSTM cell.
+
+    Args:
+      num_units: int, The number of hidden units of each cell state
+        and hidden state.
+      depth: int, The number of layers in the nest
+      forget_bias: float, The bias added to forget gates.
+      state_is_tuple: If True, accepted and returned states are tuples of
+        the `h_state` and `c_state`s.  If False, they are concatenated
+        along the column axis.  The latter behavior will soon be deprecated.
+      activation: Activation function of the inner states.  Default: `tanh`.
+      reuse: (optional) Python boolean describing whether to reuse variables
+        in an existing scope.  If not `True`, and the existing scope already has
+        the given variables, an error is raised.
+      name: String, the name of the layer. Layers with the same name will
+        share weights, but to avoid mistakes we require reuse=True in such
+        cases.
+    """
+    super(NLSTMCell, self).__init__(_reuse=reuse, name=name)
+    if not state_is_tuple:
+      logging.warn("%s: Using a concatenated state is slower and will soon be "
+                   "deprecated.  Use state_is_tuple=True.", self)
+
+    # Inputs must be 2-dimensional.
+    self.input_spec = base_layer.InputSpec(ndim=2)
+    self._num_units = num_units
+    self._forget_bias = forget_bias
+    self._state_is_tuple = state_is_tuple
+    self._depth = depth
+    self._activation = activation or math_ops.tanh
+    self._kernels = None
+    self._biases = None
+    self.built = False
+
+  @property
+  def state_size(self):
```

Comment:
Please add assertions about `state_size`, `output_size` and `depth` in the unit tests.

Reply:
Done!